docs/sync-storage-handbook/src/README.md

```diff
@@ -0,0 +1,36 @@
+# Introduction
+
+The Application Services libraries provide cross-platform components for storing and syncing user data within the Firefox ecosystem. As we make Firefox available on more platforms, and ship more products that aren't web browsers, we're finding that each product wants to access existing Firefox data, like bookmarks, history, and logins, and make its own data available to others. The goal of the sync and storage components is to expose a uniform, flexible, and high-level way to do this.
+
+* High-level means your application thinks in terms of _what_ it wants to do—"add a bookmark", "store a page visit", "update a saved password", "sync now"—and not _how_. The component takes care of details like defining the database schema, handling migrations, optimizing queries, marshaling and unmarshaling Sync records, and resolving merge conflicts.
+* Uniform means one way to access data everywhere. The same building blocks are used in logins, bookmarks, and history, as well as any custom data types. These are also the same across products and platforms, so you don't need to change three different codebases.
+* Flexible means it's easy for your application to add new data types, evolve its storage schema, and experiment with new ways to represent data.
+
+## Why?
+
+Historically, each product had to build its own storage and sync system. They often started with similar data models, but then evolved based on immediate product needs. Changes had to be backported to each platform, often across languages: Firefox Desktop was written in a mix of JavaScript and C++, Firefox for Android in Java, and Firefox for iOS in Swift.
+
+Beyond the language barrier, there was little commonality between the implementations. Some concepts didn't translate well, if at all, and coordinating changes across platforms was hard. Syncing was often bolted on, and required lots of low-level integration work. This made for an inconsistent, brittle developer experience that affected the final product.
```

Comment:
I think it would be fair to say it lead to "an inconsistent, brittle *user* experience" as well.